deviceModeInit.js

```diff
@@ -337,7 +452,20 @@ var rudderTracking = (function () {
         }
       } else {
         if (payload[j.src]) {
-          destinationPayload[j.dest] = payload[j.src];
+          // If there are many variants for a product then we will be sending data for the one currently visible 
+          if (j.src === "variants" && Array.isArray(payload[j.src])) {
```

Comment:
This change handles to select a particular variant object instead of all variants objects array

Reply:
in RudderStack variant is supported as a `string`. We can do Shopify `variant title` for currently viewed variant to RudderSack `variant` property mapping

Reply:
We will take care of this in ecom spec changes for Shopify